### PR TITLE
docs: include types from common mdx chunks

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

Add types of api data in documents. Each type should be a collapse panel and can expand by clicking, just like below:

image

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: website/components/Collapse.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.ant-collapse-header {
+  padding: 6px 10px !important;
```
Comment (chenjiahan) [inline website/components/Collapse.scss:2]: the original padding of antd looks better..

File: website/docs/en/types/asset-info.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<>
```
Comment (chenjiahan) [inline website/docs/en/types/asset-info.mdx]: why do we need this fragment, it looks wired

File: website/docs/en/types/cache.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        computer: (fn: (err: Error, result: T) => void) => void,
+        callback: (err: Error, result: T) => void,
+      ): void;
+      providePromise<T>(
+        computer: (fn: (err: Error, result: T) => void) => void,
+      ): Promise<T>;
+};
+
+type CacheFacade = {
+getChildCache(name: string): CacheFacade;
```
Comment (chenjiahan) [inline website/docs/en/types/cache.mdx]: the ident is broken

File: website/docs/en/api/javascript-api.mdx [hunk 4]
```diff
@@ -66,6 +70,16 @@ The `run` method is then used to kickstart all compilation work. Upon completion
 The API only supports a single concurrent compilation at a time. When using `run` or `watch`, call `close` and wait for it to finish before calling `run` or `watch` again. Concurrent compilations will corrupt the output files.
 :::
 
+<Collapse>
+  <CollapsePanel
+    className="collapse-code-panel"
```
Comment (chenjiahan) [inline website/docs/en/api/javascript-api.mdx:75]: `collapse-code-panel` repeated many times, can we simplify it?

For example, replace it with `type="code"` or something else

File: website/components/Collapse.scss (hunk 1 above)
Comment (LingyuCoder) [reply website/components/Collapse.scss:2]: Too big and will use too much space

File: website/docs/en/types/asset-info.mdx (hunk 2 above)
Comment (LingyuCoder) [reply website/docs/en/types/asset-info.mdx]: bug of rspress

File: website/docs/en/types/cache.mdx (hunk 3 above)
Comment (LingyuCoder) [reply website/docs/en/types/cache.mdx]: fixed, david`s solution not works

File: website/docs/en/api/javascript-api.mdx (hunk 4 above)
Comment (LingyuCoder) [reply website/docs/en/api/javascript-api.mdx:75]: this is original collapse antd component, class name is just to override its style of padding and margin. we do not need to wrap it as a new component

**Pull request merged**